motion/motion.go

```diff
@@ -73,29 +73,27 @@ type motionDetector struct {
 	count             int
 	background        [lepton3.FrameRows][lepton3.FrameCols]uint16
 	backgroundWeight  float32
+	backgroundFrames  int
 	debug             *debugTracker
 	previewFrames     int
 }
 
 func (d *motionDetector) calculateThreshold(backAverage float64) {
-	oldTemp := d.tempThresh
 	d.tempThresh = uint16(math.Min(backAverage, float64(d.defaultTempThresh)))
-	if oldTemp != d.tempThresh {
-		d.debug.update("thresh", int(d.tempThresh))
-	}
 }
 
 func (d *motionDetector) Detect(frame *lepton3.Frame) bool {
-	if d.dynamicThresh {
+	if d.dynamicThresh && !isAffectedByFFC(frame) {
 		backAverage, changed := d.updateBackground(frame)
-		if changed && d.count > d.previewFrames {
+		if changed && d.backgroundFrames > d.previewFrames {
 			d.calculateThreshold(backAverage)
 			d.backgroundWeight = frameBackgroundWeighting
 		} else {
 			if d.count%weightEveryNFrames == 0 {
 				d.backgroundWeight = d.backgroundWeight * frameBackgroundWeighting
 			}
 		}
+		d.debug.update("thresh", int(d.tempThresh))
```

Comment:
what happens here when `d.debug` is `nil`?

Reply:
Handled in the debugtracker.go

Reply:
Yay past Menno :)